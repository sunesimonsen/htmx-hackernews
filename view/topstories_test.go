package view

import (
	"errors"
	"testing"

	"github.com/alecthomas/assert/v2"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/sunesimonsen/htmx-hackernews/mock"
	"github.com/sunesimonsen/htmx-hackernews/templates"
)

func TestTopStoriesView(t *testing.T) {
	renderer := templates.NewRenderer()

	t.Run("when the repository returns an error", func(t *testing.T) {
		testerr := errors.New("test")
		repo := mock.TopStoryIdsRepo{Err: testerr}
		view := TopStoriesView{Repo: repo, Templates: renderer}
		_, err := view.Render(mock.Params{}, mock.Headers{})
		assert.Equal(t, err, testerr)
	})

	t.Run("when the view renders succesfully", func(t *testing.T) {
		ids := []int{42, 3545, 345, 1}
		repo := mock.TopStoryIdsRepo{Ids: ids}
		view := TopStoriesView{Repo: repo, Templates: renderer}

		data, err := view.Render(mock.Params{}, mock.Headers{})

		assert.NoError(t, err)
		snaps.MatchSnapshot(t, string(data))
	})
}
