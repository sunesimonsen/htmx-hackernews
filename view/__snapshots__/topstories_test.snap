
[TestTopStoriesView/when_the_view_renders_succesfully - 1]

    <article class="story" hx-get="/story/42?embed" hx-trigger="revealed" hx-swap="outerHTML">
        <a class="title"><div class="skeleton" style="width: 60%"></div></a>
        <footer><div class="skeleton" style="width: 40%"></div></footer>
    </article>

    <article class="story" hx-get="/story/3545?embed" hx-trigger="revealed" hx-swap="outerHTML">
        <a class="title"><div class="skeleton" style="width: 60%"></div></a>
        <footer><div class="skeleton" style="width: 40%"></div></footer>
    </article>

    <article class="story" hx-get="/story/345?embed" hx-trigger="revealed" hx-swap="outerHTML">
        <a class="title"><div class="skeleton" style="width: 60%"></div></a>
        <footer><div class="skeleton" style="width: 40%"></div></footer>
    </article>

    <article class="story" hx-get="/story/1?embed" hx-trigger="revealed" hx-swap="outerHTML">
        <a class="title"><div class="skeleton" style="width: 60%"></div></a>
        <footer><div class="skeleton" style="width: 40%"></div></footer>
    </article>


---
